src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeColorization.cs

```diff
@@ -113,44 +113,52 @@ internal static void ApplyTheme()
     {
         string themeToApply = GetSystemTheme();
 
-        Window currentWindow = Application.Current.MainWindow;
+        foreach (Window window in Application.Current.Windows)
+        {        
+            ApplyTheme(window);
+        }
+
+        _currentApplicationTheme = themeToApply;
+        _currentAppsTheme = GetAppsTheme();
+    }
+
+    internal static void ApplyTheme(Windows.Window window)
+    {
+        string themeToApply = GetSystemTheme();
 
         if (IsThemeDark() && Utility.IsOSWindows11OrNewer)
         {
             UpdateApplicationResources(new Uri("pack://application:,,,/PresentationFramework.FluentWindows;component/Resources/Theme/" + "dark.xaml", UriKind.Absolute));
```

Comment:
We should have this in `ApplyTheme()` method. Updating resources multiple times for same theme switch will have perf issues.

Reply:
I have refactored this method all together. Along with this I have refactored the themedictionary addition in Window.cs, I noticed that, the same dictionary was being added multiple times when we were opening multiple windows.